### ISSUE TITLE
tensorflow入门-mnist手写数字识别(一，网络搭建) | 极客兔兔

### ISSUE THREAD

**Issue by geektutu** [geektutu/geektutu-blog#4]
https://geektutu.com/post/tensorflow-mnist-simplest.html 

 tensorflow入门系列文章，mnist手写数字识别(一，网络搭建)。

**Comment by qiuChengleiy** [comment]
mark 一下

**Comment by SauryNiu** [comment]
计算交叉熵的时候，-log(0.6)如果是默认底数为10的话，不是应该等于0.22吗

**Comment by geektutu** [comment]
@SauryNiu 编程语言里，一般默认以e为底数。

**Comment by jinge1001** [comment]
请问激活函数主要有什么作用呀？

**Comment by geektutu** [comment]
@jinge1001 空间映射，引入非线性因素，弥补线性模型不能解决的问题。映射后保证数据可求偏导，且保留了原始数据的特征。

**Comment by jinge1001** [comment]
> @geektutu
> @jinge1001 空间映射，引入非线性因素，弥补线性模型不能解决的问题。映射后保证数据可求偏导，且保留了原始数据的特征。
> 

好的，那选哪种激活函数主要依据什么呀？

**Comment by ZeJion** [comment]
你好 我使用的是2.0.0版本 为什么model里面tf.train.GradientDescentOptimizer显示不存在还有log也显示不存在

**Comment by ZeJion** [comment]
我将tf.train改为tf.keras.train好像就可以了 这是为什么

**Comment by King-Key** [comment]
star

**Comment by Vladimir-Chan** [comment]
博主，这是tensorflow 2.0的教程吗，2.0版本应该已经没有placeholder这些函数了啊

**Comment by geektutu** [comment]
@Vladimir-Chan，前四篇是1.x 版本的时候写的，第五篇开始是2.0开始写的。

**Comment by heat3wade** [comment]
楼主您好，非常感谢您的分享，请问一下，这个结果大概运行之后多久能跑出来？我运行之后pycharm输出框里面就少都没有。。。python是3.6.4的，tensorflow是1.2.1的

**Comment by wangzaistone** [comment]
in mac run it  ,report error,which add as  follows 
import os 
os.environ['KMP_DUPLICATE_LIB_OK']='True'

it works 
refers:https://github.com/dmlc/xgboost/issues/1715

**Comment by hyuan-kamuda** [comment]
交叉熵损失函数不需要自己定义吧，tf都提供了。

1. tf.nn.softmax_cross_entropy_with_logits(logits, labels))
logits是神经网络预测值（未经softmax），labels是真实Y的概率值（一般为one-hot编码）

2. tf.nn.sparse_softmax_cross_entropy_with_logits(logits, labels))
logits是神经网络预测值（未经softmax），labels是真实Y的位置码
